From: Brent Baude <bbaude@redhat.com>
Date: Mon, 25 Mar 2024 13:17:37 -0500
Subject: Fix reference to deprecated types.Info

upstream docker folks deprecated types.Info in favor of system.Info.

fixes #22132 and #22117

Signed-off-by: Brent Baude <bbaude@redhat.com>
---
 pkg/api/handlers/compat/info.go | 1 -
 pkg/api/handlers/types.go       | 5 ++++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/pkg/api/handlers/compat/info.go b/pkg/api/handlers/compat/info.go
index 94b7bdf..1822f57 100644
--- a/pkg/api/handlers/compat/info.go
+++ b/pkg/api/handlers/compat/info.go
@@ -51,7 +51,6 @@ func GetInfo(w http.ResponseWriter, r *http.Request) {
 
 	// FIXME: Need to expose if runtime supports Checkpointing
 	// liveRestoreEnabled := criu.CheckForCriu() && configInfo.RuntimeSupportsCheckpoint()
-
 	info := &handlers.Info{
 		Info: dockerSystem.Info{
 			Architecture:       goRuntime.GOARCH,
diff --git a/pkg/api/handlers/types.go b/pkg/api/handlers/types.go
index 194a8d3..5a13530 100644
--- a/pkg/api/handlers/types.go
+++ b/pkg/api/handlers/types.go
@@ -7,6 +7,7 @@ import (
 	dockerContainer "github.com/docker/docker/api/types/container"
 	dockerNetwork "github.com/docker/docker/api/types/network"
 	"github.com/docker/docker/api/types/registry"
+	"github.com/docker/docker/api/types/system"
 	"github.com/opencontainers/runtime-spec/specs-go"
 )
 
@@ -16,6 +17,8 @@ type AuthConfig struct {
 
 type ImageInspect struct {
 	docker.ImageInspect
+	// Container is for backwards compat but is basically unused
+	Container string
 }
 
 type ContainerConfig struct {
@@ -74,7 +77,7 @@ type UpdateEntities struct {
 }
 
 type Info struct {
-	docker.Info
+	system.Info
 	BuildahVersion     string
 	CPURealtimePeriod  bool
 	CPURealtimeRuntime bool
